diff -uNr luci-0.23.0__orig/luci/lib/ricci_communicator.py luci-0.23.0/luci/lib/ricci_communicator.py
--- luci-0.23.0__orig/luci/lib/ricci_communicator.py	2011-11-21 00:48:55.304523392 +0000
+++ luci-0.23.0/luci/lib/ricci_communicator.py	2011-11-21 00:56:00.685509595 +0000
@@ -674,10 +674,16 @@
 def resolve_cluster_version(uname_str):
     if 'Santiago' in uname_str:
         return (3, 'RHEL', uname_str[uname_str.find('6.'):].split(' ')[0])
+    elif 'CentOS release 6.' in uname_str:
+        return (3, 'RHEL', uname_str[uname_str.find('6.'):].split(' ')[0])
     elif 'Tikanga' in uname_str:
         return (2, 'RHEL', uname_str[uname_str.find('5.'):].split(' ')[0])
+    elif 'CentOS release 5.' in uname_str:
+        return (2, 'RHEL', uname_str[uname_str.find('5.'):].split(' ')[0])
     elif 'Nahant' in uname_str:
         return (1, 'RHEL', uname_str[uname_str.find('4.'):].split(' ')[0])
+    elif 'CentOS release 4.' in uname_str:
+        return (1, 'RHEL', uname_str[uname_str.find('4.'):].split(' ')[0])
     elif 'Lovelock' in uname_str:
         return (3, 'Fedora', '15')
     elif 'Laughlin' in uname_str:
