diff -up mesa-11.0.7/include/EGL/eglext.h.dave mesa-11.0.7/include/EGL/eglext.h
--- mesa-11.0.7/include/EGL/eglext.h.dave	2015-11-28 12:37:59.000000000 -0500
+++ mesa-11.0.7/include/EGL/eglext.h	2015-12-09 19:19:17.654503223 -0500
@@ -880,7 +880,6 @@ EGLAPI EGLuint64NV EGLAPIENTRY eglGetSys
 #endif /* EGL_TIZEN_image_native_surface */
 
 #include <EGL/eglmesaext.h>
-#include <EGL/eglextchromium.h>
 
 #ifdef __cplusplus
 }
diff -up mesa-11.0.7/src/egl/drivers/dri2/egl_dri2.c.dave mesa-11.0.7/src/egl/drivers/dri2/egl_dri2.c
--- mesa-11.0.7/src/egl/drivers/dri2/egl_dri2.c.dave	2015-12-07 08:59:47.000000000 -0500
+++ mesa-11.0.7/src/egl/drivers/dri2/egl_dri2.c	2015-12-09 19:19:17.654503223 -0500
@@ -1513,15 +1513,6 @@ dri2_create_image_wayland_wl_buffer(_EGL
 }
 #endif
 
-static EGLBoolean
-dri2_get_sync_values_chromium(_EGLDisplay *dpy, _EGLSurface *surf,
-                              EGLuint64KHR *ust, EGLuint64KHR *msc,
-                              EGLuint64KHR *sbc)
-{
-   struct dri2_egl_display *dri2_dpy = dri2_egl_display(dpy);
-   return dri2_dpy->vtbl->get_sync_values(dpy, surf, ust, msc, sbc);
-}
-
 /**
  * Set the error code after a call to
  * dri2_egl_image::dri_image::createImageFromTexture.
@@ -2526,7 +2517,6 @@ _eglBuiltInDriverDRI2(const char *args)
    dri2_drv->base.API.UnbindWaylandDisplayWL = dri2_unbind_wayland_display_wl;
    dri2_drv->base.API.QueryWaylandBufferWL = dri2_query_wayland_buffer_wl;
 #endif
-   dri2_drv->base.API.GetSyncValuesCHROMIUM = dri2_get_sync_values_chromium;
    dri2_drv->base.API.CreateSyncKHR = dri2_create_sync;
    dri2_drv->base.API.ClientWaitSyncKHR = dri2_client_wait_sync;
    dri2_drv->base.API.WaitSyncKHR = dri2_server_wait_sync;
diff -up mesa-11.0.7/src/egl/drivers/dri2/platform_x11.c.dave mesa-11.0.7/src/egl/drivers/dri2/platform_x11.c
--- mesa-11.0.7/src/egl/drivers/dri2/platform_x11.c.dave	2015-12-07 08:59:47.000000000 -0500
+++ mesa-11.0.7/src/egl/drivers/dri2/platform_x11.c	2015-12-09 19:19:17.655503216 -0500
@@ -1314,7 +1314,6 @@ dri2_initialize_x11_dri2(_EGLDriver *drv
    disp->Extensions.NOK_swap_region = EGL_TRUE;
    disp->Extensions.NOK_texture_from_pixmap = EGL_TRUE;
    disp->Extensions.NV_post_sub_buffer = EGL_TRUE;
-   disp->Extensions.CHROMIUM_sync_control = EGL_TRUE;
 
 #ifdef HAVE_WAYLAND_PLATFORM
    disp->Extensions.WL_bind_wayland_display = EGL_TRUE;
diff -up mesa-11.0.7/src/egl/main/eglapi.c.dave mesa-11.0.7/src/egl/main/eglapi.c
--- mesa-11.0.7/src/egl/main/eglapi.c.dave	2015-12-09 19:19:17.655503216 -0500
+++ mesa-11.0.7/src/egl/main/eglapi.c	2015-12-09 19:24:34.089275514 -0500
@@ -383,8 +383,6 @@ _eglCreateExtensionsString(_EGLDisplay *
    /* Please keep these sorted alphabetically. */
    _EGL_CHECK_EXTENSION(ANDROID_image_native_buffer);
 
-   _EGL_CHECK_EXTENSION(CHROMIUM_sync_control);
-
    _EGL_CHECK_EXTENSION(EXT_buffer_age);
    _EGL_CHECK_EXTENSION(EXT_create_context_robustness);
    _EGL_CHECK_EXTENSION(EXT_image_dma_buf_import);
@@ -1734,28 +1732,6 @@ eglPostSubBufferNV(EGLDisplay dpy, EGLSu
 }
 
 static EGLBoolean EGLAPIENTRY
-eglGetSyncValuesCHROMIUM(EGLDisplay display, EGLSurface surface,
-                         EGLuint64KHR *ust, EGLuint64KHR *msc,
-                         EGLuint64KHR *sbc)
-{
-   _EGLDisplay *disp = _eglLockDisplay(display);
-   _EGLSurface *surf = _eglLookupSurface(surface, disp);
-   _EGLDriver *drv;
-   EGLBoolean ret;
-
-   _EGL_CHECK_SURFACE(disp, surf, EGL_FALSE, drv);
-   if (!disp->Extensions.CHROMIUM_sync_control)
-      RETURN_EGL_EVAL(disp, EGL_FALSE);
-
-   if (!ust || !msc || !sbc)
-      RETURN_EGL_ERROR(disp, EGL_BAD_PARAMETER, EGL_FALSE);
-
-   ret = drv->API.GetSyncValuesCHROMIUM(disp, surf, ust, msc, sbc);
-
-   RETURN_EGL_EVAL(disp, ret);
-}
-
-static EGLBoolean EGLAPIENTRY
 eglExportDMABUFImageQueryMESA(EGLDisplay dpy, EGLImage image,
                               EGLint *fourcc, EGLint *nplanes,
                               EGLuint64KHR *modifiers)
@@ -1874,7 +1850,6 @@ eglGetProcAddress(const char *procname)
       { "eglGetPlatformDisplayEXT", (_EGLProc) eglGetPlatformDisplayEXT },
       { "eglCreatePlatformWindowSurfaceEXT", (_EGLProc) eglCreatePlatformWindowSurfaceEXT },
       { "eglCreatePlatformPixmapSurfaceEXT", (_EGLProc) eglCreatePlatformPixmapSurfaceEXT },
-      { "eglGetSyncValuesCHROMIUM", (_EGLProc) eglGetSyncValuesCHROMIUM },
       { "eglExportDMABUFImageQueryMESA", (_EGLProc) eglExportDMABUFImageQueryMESA },
       { "eglExportDMABUFImageMESA", (_EGLProc) eglExportDMABUFImageMESA },
       { NULL, NULL }
diff -up mesa-11.0.7/src/egl/main/eglapi.h.dave mesa-11.0.7/src/egl/main/eglapi.h
--- mesa-11.0.7/src/egl/main/eglapi.h.dave	2015-11-28 12:37:59.000000000 -0500
+++ mesa-11.0.7/src/egl/main/eglapi.h	2015-12-09 19:25:20.013952498 -0500
@@ -118,8 +118,6 @@ typedef EGLint (*QueryBufferAge_t)(_EGLD
 
 typedef EGLBoolean (*SwapBuffersWithDamageEXT_t) (_EGLDriver *drv, _EGLDisplay *dpy, _EGLSurface *surface, const EGLint *rects, EGLint n_rects);
 
-typedef EGLBoolean (*GetSyncValuesCHROMIUM_t) (_EGLDisplay *dpy, _EGLSurface *surface, EGLuint64KHR *ust, EGLuint64KHR *msc, EGLuint64KHR *sbc);
-
 typedef EGLBoolean (*ExportDMABUFImageQueryMESA_t)(_EGLDriver *drv, _EGLDisplay *disp, _EGLImage *img, EGLint *fourcc, EGLint *nplanes, EGLuint64KHR *modifiers);
 typedef EGLBoolean (*ExportDMABUFImageMESA_t)(_EGLDriver *drv, _EGLDisplay *disp, _EGLImage *img, EGLint *fds, EGLint *strides, EGLint *offsets);
 
@@ -184,7 +182,6 @@ struct _egl_api
    PostSubBufferNV_t PostSubBufferNV;
 
    QueryBufferAge_t QueryBufferAge;
-   GetSyncValuesCHROMIUM_t GetSyncValuesCHROMIUM;
 
    ExportDMABUFImageQueryMESA_t ExportDMABUFImageQueryMESA;
    ExportDMABUFImageMESA_t ExportDMABUFImageMESA;
diff -up mesa-11.0.7/src/egl/main/egldisplay.h.dave mesa-11.0.7/src/egl/main/egldisplay.h
--- mesa-11.0.7/src/egl/main/egldisplay.h.dave	2015-12-09 19:19:22.036472349 -0500
+++ mesa-11.0.7/src/egl/main/egldisplay.h	2015-12-09 19:26:29.365464844 -0500
@@ -92,8 +92,6 @@ struct _egl_extensions
    /* Please keep these sorted alphabetically. */
    EGLBoolean ANDROID_image_native_buffer;
 
-   EGLBoolean CHROMIUM_sync_control;
-
    EGLBoolean EXT_buffer_age;
    EGLBoolean EXT_create_context_robustness;
    EGLBoolean EXT_image_dma_buf_import;
diff -up mesa-11.0.7/src/egl/Makefile.am.dave mesa-11.0.7/src/egl/Makefile.am
--- mesa-11.0.7/src/egl/Makefile.am.dave	2015-12-09 19:27:57.730843726 -0500
+++ mesa-11.0.7/src/egl/Makefile.am	2015-12-09 19:28:44.313516402 -0500
@@ -107,7 +107,6 @@ egldir = $(includedir)/EGL
 egl_HEADERS = \
 	$(top_srcdir)/include/EGL/eglext.h \
 	$(top_srcdir)/include/EGL/egl.h \
-	$(top_srcdir)/include/EGL/eglextchromium.h \
 	$(top_srcdir)/include/EGL/eglmesaext.h \
 	$(top_srcdir)/include/EGL/eglplatform.h
 
