diff -uNrp libreport-2.0.9.orig/src/plugins/bugzilla.conf libreport-2.0.9/src/plugins/bugzilla.conf
--- libreport-2.0.9.orig/src/plugins/bugzilla.conf	2011-11-14 14:36:39.000000000 +0000
+++ libreport-2.0.9/src/plugins/bugzilla.conf	2013-01-31 21:35:16.505783245 +0000
@@ -1,5 +1,5 @@
 # Bugzilla URL
-BugzillaURL = https://bugzilla.redhat.com/
+BugzillaURL = https://bugzilla.example.com/
 # yes means that ssl certificates will be checked
 SSLVerify = yes
 # your login has to exist, if you don have any, please create one
