diff -uNrp abrt-2.0.8.orig/src/daemon/gpg_keys abrt-2.0.8/src/daemon/gpg_keys
--- abrt-2.0.8.orig/src/daemon/gpg_keys	2011-11-16 12:25:01.000000000 +0000
+++ abrt-2.0.8/src/daemon/gpg_keys	2012-06-22 18:11:51.074742677 +0000
@@ -1 +1,4 @@
-/etc/pki/rpm-gpg/RPM-GPG-KEY-fedora
+/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-6
+/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Debug-6
+/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Security-6
+/etc/pki/rpm-gpg/RPM-GPG-KEY-CentOS-Testing-6
