diff -uNrp freeipa-3.3.3.orig/ipaserver/install/ntpinstance.py freeipa-3.3.3/ipaserver/install/ntpinstance.py
--- freeipa-3.3.3.orig/ipaserver/install/ntpinstance.py	2013-11-01 10:34:30.000000000 -0500
+++ freeipa-3.3.3/ipaserver/install/ntpinstance.py	2014-06-26 07:27:19.644718099 -0500
@@ -43,6 +43,8 @@ class NTPInstance(service.Service):
         os = ""
         if ipautil.file_exists("/etc/fedora-release"):
             os = "fedora"
+        elif ipautil.file_exists("/etc/centos-release"):
+            os = "centos"
         elif ipautil.file_exists("/etc/redhat-release"):
             os = "rhel"
 
