diff --git a/src/mesa/drivers/dri/i965/intel_screen.c b/src/mesa/drivers/dri/i965/intel_screen.c
index 6618c1a..09a575b 100644
--- a/src/mesa/drivers/dri/i965/intel_screen.c
+++ b/src/mesa/drivers/dri/i965/intel_screen.c
@@ -1109,20 +1109,9 @@ intel_detect_swizzling(struct intel_screen *screen)
 const int*
 intel_supported_msaa_modes(const struct intel_screen  *screen)
 {
-   static const int gen8_modes[] = {8, 4, 2, 0, -1};
-   static const int gen7_modes[] = {8, 4, 0, -1};
-   static const int gen6_modes[] = {4, 0, -1};
    static const int gen4_modes[] = {0, -1};
 
-   if (screen->devinfo->gen >= 8) {
-      return gen8_modes;
-   } else if (screen->devinfo->gen >= 7) {
-      return gen7_modes;
-   } else if (screen->devinfo->gen == 6) {
-      return gen6_modes;
-   } else {
-      return gen4_modes;
-   }
+   return gen4_modes;
 }
 
 static __DRIconfig**
@@ -1235,11 +1224,6 @@ intel_screen_make_configs(__DRIscreen *dri_screen)
          stencil_bits[1] = 8;
       }
 
-      if (devinfo->gen >= 7)
-         num_msaa_modes = 2;
-      else if (devinfo->gen == 6)
-         num_msaa_modes = 1;
-
       new_configs = driCreateConfigs(formats[i],
                                      depth_bits,
                                      stencil_bits,
