diff -uNr kde-settings-4.3.1__orig/etc/kde/kdm/kdmrc kde-settings-4.3.1/etc/kde/kdm/kdmrc
--- kde-settings-4.3.1__orig/etc/kde/kdm/kdmrc	2010-04-26 13:32:39.000000000 +0000
+++ kde-settings-4.3.1/etc/kde/kdm/kdmrc	2011-07-01 22:11:57.000000000 +0000
@@ -438,7 +438,7 @@
 UseTheme=true
 # The theme to use for the greeter. Can point to either a directory or an XML
 # file. Default is "/usr/share/kde4/apps/kdm/themes/circles"
-Theme=/usr/share/kde4/apps/kdm/themes/RHEL6
+Theme=/usr/share/kde4/apps/kdm/themes/CentOS6
 # Set the widget color scheme 
 ColorScheme=Oxygen
 
diff -uNr kde-settings-4.3.1__orig/usr/share/kde-settings/kde-profile/default/share/config/ksplashrc kde-settings-4.3.1/usr/share/kde-settings/kde-profile/default/share/config/ksplashrc
--- kde-settings-4.3.1__orig/usr/share/kde-settings/kde-profile/default/share/config/ksplashrc	2010-04-22 12:46:07.000000000 +0000
+++ kde-settings-4.3.1/usr/share/kde-settings/kde-profile/default/share/config/ksplashrc	2011-07-01 22:11:06.000000000 +0000
@@ -1,2 +1,2 @@
 [KSplash]
-Theme=RHEL6
+Theme=CentOS6
diff -uNr kde-settings-4.3.1__orig/usr/share/kde-settings/kde-profile/default/share/config/plasma-desktop-appletsrc kde-settings-4.3.1/usr/share/kde-settings/kde-profile/default/share/config/plasma-desktop-appletsrc
--- kde-settings-4.3.1__orig/usr/share/kde-settings/kde-profile/default/share/config/plasma-desktop-appletsrc	2010-04-22 12:46:07.000000000 +0000
+++ kde-settings-4.3.1/usr/share/kde-settings/kde-profile/default/share/config/plasma-desktop-appletsrc	2011-07-01 22:10:24.000000000 +0000
@@ -1,4 +1,4 @@
 [Containments][1][Wallpaper][image]
-userswallpapers=/usr/share/wallpapers/RHEL6/
-wallpaper=/usr/share/wallpapers/RHEL6/
+userswallpapers=/usr/share/wallpapers/CentOS6/
+wallpaper=/usr/share/wallpapers/CentOS6/
 wallpaperposition=2
